lib/dir.go

```diff
@@ -80,6 +83,62 @@ func CopyThing(srcpath, destpath string) error {
 	if srcInfo.IsDir() {
 		return DirCopy(destpath, srcpath)
 	} else {
-		return FileCopy(destpath, srcpath)
+		return FileCopy(destpath, srcpath, nil, -1, -1)
 	}
 }
+
+// Chmod changes file permissions
+func Chmod(mode, destpath string) error {
+	destInfo, err := os.Lstat(destpath)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	if destInfo.IsDir() {
```

Comment:
Sorry, I know you said some of the fixes broke things, not sure if this was one of them - but why should chmod on a directory not be allowed?

Reply:
What is the semantics of a directory as a target? its subdirs and files? what about their mode/ownership?